vortex-array/src/array/chunked/mod.rs

```diff
@@ -132,6 +133,77 @@ impl ChunkedArray {
     pub fn array_stream(&self) -> impl ArrayStream + '_ {
         ArrayStreamAdapter::new(self.dtype().clone(), stream::iter(self.chunks().map(Ok)))
     }
+
+    pub fn rechunk_default(&self) -> VortexResult<Self> {
+        let gibibyte = 1 << 30;
```

Comment:
Should this instead match the ideal IO unit, which for S3 is 16MB?

Reply:
Yeah, that probably makes more sense. I'm still trying to get my hands on the implications of `ChunkedArray`. It seems like a whole ChunkedArray is currently resident in RAM, but it also seems like a temporary situation. Definitely makes sense to me that we want IO unit sized chunks if the `ChunkedArray` is more like a stream.

Reply:
There are 3 things imho. There's a ChunkedAray, Stream<Array> and ChunkedLayout. ChunkedArray will stay all in memory